kmir/src/kmir/__main__.py

```diff
@@ -80,7 +81,12 @@ class ProveViewOpts(KMirOpts):
 def _kmir_run(opts: RunOpts) -> None:
     tools = haskell_semantics() if opts.haskell_backend else llvm_semantics()
 
-    parse_result = parse_json(tools.definition, opts.input_file, 'Pgm')
+    cargo = CargoPackage(Path.cwd())
+
+    target = opts.bin if opts.bin else cargo.default_target
+    smir_file = cargo.smir_for(target)
```

Comment:
This is where the `--file` option would be implemented: if given, `smir_file is taken from there instead of asking `cargo`.